### PR TITLE
Allow to disable the use of silent auth when using Refresh Tokens

### PR DESCRIPTION
**Pull request by frederikprijck**
When using our SDK in environments such as electron, where it's common to use the `file:` protocol, using silent auth (iframes) isn't supported. This is because of the fact that electron will set the window's `origin` to `file://`, which is a value not supported by Auth0.

This PR adds a flag, useRefreshTokensFallback, that, when set to `false`, allows disabling the use of silent auth when using Refresh Tokens. The flag defaults to `true`.

The implication here is that, as we won't fall back to iframes, the errors regarding refresh tokens (`missing_refresh_token` or `invalid_grant`) will be thrown to the end-user, needing them to catch these errors and send the user through an interactive login.

```ts
let token: string;
try {
  token = await auth0.getTokenSilently();
} catch (e) {
  if (e.error === 'missing_refresh_token' || e.error === 'invalid_grant') {
    auth0.loginWithRedirect();
  }
}
```

### CODE REVIEW

File: src/Auth0Client.ts [hunk 1]
```diff
@@ -299,6 +305,11 @@ export default class Auth0Client {
     }
 
     this.customOptions = getCustomInitialOptions(options);
+
+    this.useRefreshTokensFallback =
```
Comment (adamjmcgrath) [inline src/Auth0Client.ts:309]: `this.useRefreshTokensFallback = this.options.useRefreshTokensFallback !== false;`

File: src/global.ts [hunk 2]
```diff
@@ -158,6 +158,26 @@ export interface Auth0ClientOptions extends BaseLoginOptions {
    */
   useRefreshTokens?: boolean;
 
+  /**
+   * If true, fallback to the legacy technique of using a hidden iframe and the `authorization_code` grant with `prompt=none` when unable to use refresh tokens.
```
Comment (adamjmcgrath) [inline src/global.ts:162]: I wouldn't describe `web_message` as 'legacy'

File: src/global.ts [hunk 3]
```diff
@@ -158,6 +158,26 @@ export interface Auth0ClientOptions extends BaseLoginOptions {
...
+   * The default setting is `true`.
+   *
+   * **Note**: There might be situations where using the legacy iframe technique is not supported, and using it as a fallback might be undesired.
```
Comment (adamjmcgrath) [inline src/global.ts:165]: "There might be situations where doing silent auth with a Web Message response from an iframe is not possible, like when you're serving your application from the file system or a custom protocol (like in a Desktop or Native app). In situations like this you can disable the iframe fallback and handle the failed Refresh Grant and prompt the user to login interactively with `loginWithRedirect` or `loginWithPopup`"

File: src/global.ts (hunk 2 above)
Comment (frederikprijck) [reply src/global.ts:162]: I stole it from the comment above from useRefreshTokens 😬

File: src/Auth0Client.ts (hunk 1 above)
Comment (frederikprijck) [reply src/Auth0Client.ts:309]: Oh my :D Thanks!

File: src/global.ts (hunk 2 above)
Comment (adamjmcgrath) [reply src/global.ts:162]: Haha, I bet that was me...

**Review by adamjmcgrath — APPROVED** [review]

**Review by stevehobbsdev — APPROVED** [review]

**Pull request merged**